Source/plugins/IDispatcher.h

```diff
@@ -38,16 +38,16 @@ namespace WPEFramework {
 
                 enum { ID = RPC::ID_DISPATCHER_CALLBACK };
 
-                virtual Core::hresult Event(const string& event, const string& parameters) = 0;
+                virtual Core::hresult Event(const string& event, const ccstring& parameters) = 0;
                 virtual Core::hresult Error(const uint32_t channel, const uint32_t id, const uint32_t code, const string& message) = 0;
-                virtual Core::hresult Response(const uint32_t channel, const uint32_t id, const string& response) = 0;
+                virtual Core::hresult Response(const uint32_t channel, const uint32_t id, const ccstring& response) = 0;
 
                 virtual Core::hresult Subscribe(const uint32_t channel, const string& event, const string& designator) = 0;
                 virtual Core::hresult Unsubscribe(const uint32_t channel, const string& event, const string& designator) = 0;
             };
 
-            virtual Core::hresult Validate(const string& token, const string& method, const string& paramaters) const = 0;
-            virtual Core::hresult Invoke(ICallback* callback, const uint32_t channelId, const uint32_t id, const string& token, const string& method, const string& parameters, string& response /* @out */) = 0;
+            virtual Core::hresult Validate(const string& token, const string& method, const ccstring& paramaters) const = 0;
+            virtual Core::hresult Invoke(ICallback* callback, const uint32_t channelId, const uint32_t id, const string& token, const string& method, const string& ccparameters, string& ccresponse /* @out */) = 0;
```

Comment:
ccstring here!

Reply:
For 2 reasons:
1) Nice test for the Tools that require adaptions.
2) This is for linking up multiple WPEFrameworks through COMRPC, than the next step is that an attempt will be made to forward the recing JSONRPC through COMRPC to the other WPEFrmaework instance and I think.... given the RDKSevices and how the interface design is in those components, within a day we get a call that the JSONRPC can not be forwarded..
At the end, the COMRPC (Proxy Stubs) are only used if we hookup two WPEFrameworks and even than it is, under normal conditions just an additional 4 bytes (2 for parameters, 2 for response)  we need to send, so I guess it is not notices in any part of the metrological system deplyments..